community/dbms/src/main/java/org/neo4j/commandline/dbms/VersionCommand.java

```diff
@@ -0,0 +1,122 @@
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.impl.pagecache.StandalonePageCacheFactory;
+import org.neo4j.kernel.impl.store.MetaDataStore;
+import org.neo4j.kernel.impl.store.format.RecordFormatSelector;
+import org.neo4j.kernel.impl.store.format.RecordFormats;
+import org.neo4j.kernel.impl.storemigration.StoreVersionCheck;
+import org.neo4j.kernel.impl.util.Validators;
+import org.neo4j.kernel.internal.Version;
+
+import static org.neo4j.kernel.impl.store.format.RecordFormatSelector.findSuccessor;
+
+public class VersionCommand implements AdminCommand
+{
+    private static final Arguments arguments = new Arguments()
+            .withArgument( new MandatoryCanonicalPath( "store", "path-to-dir",
+                    "Path to database store to check version of." ) );
+
+    public static class Provider extends AdminCommand.Provider
+    {
+
+        public Provider()
+        {
+            super( "version" );
+        }
+
+        @Override
+        public Arguments allArguments()
+        {
+            return arguments;
+        }
+
+        @Override
+        public String summary()
+        {
+            return "Check the version of a Neo4j database store.";
+        }
+
+        @Override
+        public String description()
+        {
+            return "Checks the version of a Neo4j database store. Note that this command expects a path to a store " +
+                    "directory, for example --store=data/databases/graph.db.";
+        }
+
+        @Override
+        public AdminCommand create( Path homeDir, Path configDir, OutsideWorld outsideWorld )
+        {
+            return new VersionCommand( outsideWorld::stdOutLine );
+        }
+    }
+
+    private Consumer<String> out;
+
+    public VersionCommand( Consumer<String> out )
+    {
+        this.out = out;
+    }
+
+    @Override
+    public void execute( String[] args ) throws IncorrectUsage, CommandFailed
+    {
+        final Path storeDir = arguments.parseMandatoryPath( "store", args );
+
+        Validators.CONTAINS_EXISTING_DATABASE.validate( storeDir.toFile() );
+
+        try ( PageCache pageCache = StandalonePageCacheFactory.createPageCache( new DefaultFileSystemAbstraction() ) )
+        {
+            final String storeVersion = new StoreVersionCheck( pageCache )
+                    .getVersion( storeDir.resolve( MetaDataStore.DEFAULT_NAME ).toFile() )
+                    .orElseThrow(
+                            () -> new CommandFailed( String.format( "Could not find version metadata in store '%s'",
+                                    storeDir ) ) );
+
+            final String fmt = "%-20s%s";
+            out.accept( String.format( fmt, "Store version:", storeVersion ) );
+
+            RecordFormats format = RecordFormatSelector.selectForVersion( storeVersion );
+            out.accept( String.format( fmt, "Introduced in:", format.firstNeo4jVersion() ) );
```

Comment:
i think if we will change messages to: `Introduced in version:`, `Superseded in version:` and `Current version:` that will be more clear. WDYT?